src/nl/codefox/contextvh/tygronRequests.mod2g

```diff
@@ -8,6 +8,13 @@ module tygronRequests {
 		not(request(_,_,RequestID,_,_,_,_,_,_,_)))		
 		do popup_answer(RequestID,AnswerID). 
 	
+	% if a request is a confirmation that you are buying land 
+	forall percept(requests(List)),
+		bel(member(request("INFORMATION","BUY_LAND",_,_,[2],_,_,MultiPolygon,_,_),List),not(requestedLand(MultiPolygon,_))) do insert(requestedLand(MultiPolygon, 0)).
+	forall percept(requests(List)),
+		bel(member(request("INTERACTION","BUY_LAND",_,_,[2],_,"NO_PRICE",MultiPolygon,_,_),List),requestedLand(MultiPolygon,Counter),Counter<3,NewCounter is Counter+1) 
+		do delete(requestedLand(MultiPolygon, Counter), doneBuying(MultiPolygon)) + insert(requestedLand(MultiPolygon, NewCounter)) + adopt(doneBuying(MultiPolygon)).
```

Comment:
Are your sure this rules do not fire on requests that are already in the belief base instead of new percepts?